### PR TITLE
WIP: cleaning up io library and CSV file

### PR DESCRIPTION
**Pull request by nickhand**

### CODE REVIEW

File: nbodykit/tests/test_lab.py [hunk 1]
```diff
@@ -51,7 +48,7 @@ def test_paint(comm):
     output = "./test_paint-%d.bigfile" % comm.size
     alg.result.save(output)
 
-@MPIWorld(NTask=[2, 3, 4])
+@MPIWorld(NTask=[2, 3, 4], required=2)
```
Comment (nickhand) [inline nbodykit/tests/test_lab.py]: @rainwoodman -- without ``required=2`` this test appears to get run with NTask=1, must be a bug in mpi4py_test

File: nbodykit/io/csv.py [hunk 2]
```diff
@@ -1,17 +1,136 @@
 import numpy
-
+import os
+from pandas import read_csv 
+    
 from ..extern.six import string_types
 from . import FileType, tools
 
+class CSVPartition(object):
```
Comment (nickhand) [inline nbodykit/io/csv.py]: I'm using this class to cache the partitions through the ``value`` attribute, which calls read_csv on the string of bytes when requested

File: nbodykit/io/csv.py [hunk 3]
```diff
@@ -1,17 +1,136 @@
...
+    
+    Returns
+    -------
+    partitions : list of CSVPartition
+        list of objects storing the data content of each file partition,
+        stored as a bytestring
+    sizes : list of int
+        the list of the number of rows in each partition
+    """
+    from dask.bytes.utils import read_block
```
Comment (nickhand) [inline nbodykit/io/csv.py]: @rainwoodman this is the only dependency from dask still...it handles partitioning the file into blocks, always ending on the newline character. This function seems pretty stable to me...

**Comment by rainwoodman** [comment]
Merge?

**Pull request merged**